{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name_alphanumeric_underscore}}/deployment/batch_inference/predict.py

```diff
@@ -9,8 +9,40 @@ def predict_batch(
     Apply the model at the specified URI for batch inference on the table with name input_table_name,
     writing results to the table with name output_table_name
     """
-    table = spark_session.table(input_table_name)
+
     {% if cookiecutter.include_feature_store == "yes" %}
+    def rounded_unix_timestamp(dt, num_minutes=15):
+        """
+            Ceilings datetime dt to interval num_minutes, then returns the unix timestamp.
+        """
+        
+        nsecs = dt.minute * 60 + dt.second + dt.microsecond * 1e-6
+        delta = math.ceil(nsecs / (60 * num_minutes)) * (60 * num_minutes) - nsecs
+        return int((dt + timedelta(seconds=delta)).replace(tzinfo=timezone.utc).timestamp())
+
+    rounded_unix_timestamp_udf = udf(rounded_unix_timestamp, IntegerType())
+
+    _table = spark_session.table(input_table_name)
+    
+    table = (
+        _table.withColumn(
+            "rounded_pickup_datetime",
+            to_timestamp(
+                rounded_unix_timestamp_udf(
+                    _table["tpep_pickup_datetime"], lit(15)
+                    )
+                )
+            )
+        .withColumn(
+            "rounded_dropoff_datetime",
+            to_timestamp(
+                rounded_unix_timestamp_udf(
+                    _table["tpep_dropoff_datetime"], lit(30)
+                    )
+                )
+            )
+        )
```

Comment:
Ah sorry for not providing the pointers earlier. We have some doc here about how to generate the batch inference input file.
https://github.com/databricks/mlops-stack/blob/49d5615954d80641ed446a6608af816d9ae4d16c/%7B%7Bcookiecutter.root_dir__update_if_you_intend_to_use_monorepo%7D%7D/%7B%7Bcookiecutter.project_name_alphanumeric_underscore%7D%7D/deployment/batch_inference/README.md?plain=1#L6-L42

I feel it better to keep it that way because current feature store batch inference input is table `hive_metastore.default.taxi_scoring_sample_feature_store_inference_input`. Non-feature-store input table is `taxi_scoring_sample`. 
And user is likely to have the dbfs file `dbfs:/databricks-datasets/nyctaxi-with-zipcodes/subsampled`

I think if users have to execute the step of importing the file and saving the table, we can have all the processings there.

Reply:
gotcha, so what should the suggested change be/revert?

Reply:
@eric-golinko-db Thank you! Could you try reverting the `predict.py` and run the code in the README.md to generate the batch inference input file and see if it works?

Reply:
@mingyu89 reverting predict worked fine on the test repo https://dev.azure.com/ericgolinko/ado_feature_store/_build , committed change